### DISCUSSION TITLE
I need to see (almost) full relative file path on lualine

### DISCUSSION THREAD

**Discussion by armyers** [LazyVim/LazyVim#3010] [Q&A] [answered]
Hi all,

I code in terraform. I'm not sure how all TF coders structure their repos, directories and files, but the way we do it in my currrent and previous job to have the directory path identify the purpose of the TF stack, and the file names within each stack are always the same. For example,

```
~/code/github-repo/terraform
├── cloudwatch-logs
│   └── ecs
│       ├── applications
│       │   ├── README.md
│       │   ├── backend.tf
│       │   ├── data.tf
│       │   ├── labels.tf
│       │   ├── main.tf
│       │   ├── outputs.tf
│       │   ├── providers.tf
│       │   └── variables.tf
├── ec2
│   └── jumphost
│       ├── backend.tf
│       ├── data.tf
│       ├── labels.tf
│       ├── main.tf
│       ├── outputs.tf
│       ├── providers.tf
│       └── variables.tf
├── ecs
│   └── cluster
│       ├── common-services
│       │   ├── backend.tf
│       │   ├── data.tf
│       │   ├── labels.tf
│       │   ├── main.tf
│       │   ├── outputs.tf
│       │   ├── providers.tf
│       │   └── variables.tf
```

Multiply that by over a dozen git repos and dozens of stacks per repo.

The problem is that the default status line from lualine in lazyvim show the top level git repo and the file name.  I have changed the `filename` settings to `path = 3` so that it shows the path to the file, but when the path has greater than 3 parts, it replaces most of the path with `...` so I still am not able to see where I am. 

What I'd like to do is:

* disable to `root_dir` part of the status line completely
* change the filename part to show the relative path to the file without `...`. 

For the relative path, there are two scenarios:

* the file is not in a git repo: show the relative path to my home dir; eg,  /Users/my-login/some/path/to/filename.txt => ~/some/path/to/filename.txt
* the file is in a git repo: show the relative path to the git repo top level: eg, /Users/my-login/code/some-git-repo/path/to/filename.txt => some-git-repo/path/to/filename.txt

I've read this discussion https://github.com/LazyVim/LazyVim/discussions/2605 which was very helpful to understand how lualine is configured in lazyvim. 

So I've made similar edits to replace/modify the root_dir and pretty_path functions but I don't know enough lua yet to make it work the way I want.

I guess my question is: is it worth my trouble to try to change lazyvim's implementation of lualine, or should I just completely replace it with my own, or even try another status line? :)

Thanks again for the great nvim distro and the incredible help provided on this github discussion forum! :)

**Comment by dpetka2001** [comment, marked as answer]
The easiest thing to do would be something like
```lua
opts = function(_, opts)
  table.remove(opts.sections.lualine_c, 1)
  table.remove(opts.sections.lualine_c, #opts.sections.lualine_c)
  table.insert(opts.sections.lualine_c, {
    "filename",
    path = 3,
  })
end,
```
That only takes care of the relative path to your home dir. If you would like to extend that for git repos as well, you would have to write your own custom component.

Of course that means you lose the aesthetic niceties of `pretty_path`, but if you would like, you could just copy paste that function to your own config, make any necessary custom changes and use that instead.

**Reply by greg218** [reply]
Just a note for anyone who this didn't work for (the `path = 3`)... 

I had to use `{ "%f" }` in the lualine_c option.  

more background can be found in my linked comment...
https://github.com/nvim-lualine/lualine.nvim/issues/1139#issuecomment-2438078938

**Comment by armyers** [comment]
yep, that's exactly what I've done so far. I just need to figure out enough lua to make it not put in `...` when parts is > 3 (that's the easy part, and I do have that working), and to do the harder stuff of removing the `prefix` components of the path that I don't want/need :)

Thank you again!

**Comment by robsonjorgefernandez** [comment]
Thanks for the tip!

I did the following:
```lua
return {
  "nvim-lualine/lualine.nvim",
  event = "VeryLazy",
  enabled = true,
  opts = function(_, opts)
    opts.sections.lualine_c[4] = { "filename", path = 1 }
  end,
}
```
**Before**
![ksnip_20250402-163734](https://github.com/user-attachments/assets/fd62d32e-bd6f-49c4-964a-c7327041cdf5)

**After**
![ksnip_20250402-163424](https://github.com/user-attachments/assets/071dc62d-e617-44f6-9bb7-c8ce3570372f)

It works like a charm!

**Reply by szist** [reply]
I found that you could still use the pretty path from the default configs, there are opts that can be passed in. For example:
```lua
return {
  "nvim-lualine/lualine.nvim",
  event = "VeryLazy",
  opts = function(_, opts)
    opts.sections.lualine_c[4] = { LazyVim.lualine.pretty_path({
      length = 6,
    }) }
  end,
}
```
would only do the `...` if the path is longer than 6 segments instead of the [default 3](https://github.com/LazyVim/LazyVim/blob/25abbf546d564dc484cf903804661ba12de45507/lua/lazyvim/util/lualine.lua#L82)

And you still get the bold filename, highlight on change, read-only icon, etc.
